### ISSUE TITLE
GraphQL server fails to build the schema when a query/mutation contains a Kotlin List<Enum> parameter

### ISSUE THREAD

**Issue by lcareto** [quarkusio/quarkus#26739]
### Describe the bug

A Query/Mutation containing a Kotlin `List` of `enum` causes the GraphQL schema builder to fail.
The other scenario described below works fine.

Eg.
```kotlin
import org.eclipse.microprofile.graphql.GraphQLApi
import org.eclipse.microprofile.graphql.Query
import javax.json.bind.annotation.JsonbCreator

enum class MyEnum {
    FOO,
    BAR
}

data class Foo @JsonbCreator constructor(
    val enums: List<MyEnum>,
)

@GraphQLApi
class Example {

    @Query
    fun failsToStartWithList(myEnums: List<MyEnum>): Foo {
        return Foo(myEnums)
    }

    @Query
    fun deserializeWithArray(myEnums: Array<MyEnum>): Foo {
        return Foo(myEnums.toList())
    }

    @Query
    fun deserializeInnerList(foo: Foo): Foo {
        return foo
    }

    @Query
    fun deserializeNonEnumList(values: List<String>): List<String> {
        return values
    }
}
```

### Side issue
When downgrading to Quarkus 2.9.2.Final, the query `deserializeInnerList` deserializes the inner filed `values: List<MyEnum>` as an empty List.

<img width="1104" alt="Screen Shot 2022-07-14 at 12 52 38 PM" src="https://user-images.githubusercontent.com/29870258/179037557-9bdf6cf6-4290-4a8d-9e82-261789a320a8.png">

### Expected behavior

Expect the server to start and the `failsToStartWithList` query to bind the List<Enum> accordingly.

### Error
```
Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.smallrye.graphql.deployment.SmallRyeGraphQLProcessor#buildExecutionService threw an exception: io.smallrye.graphql.schema.SchemaBuilderException: Don't know what to do with [? extends com.example.MyEnum] of kind [WILDCARD_TYPE]
```

### Reproducer:
[graphql-schema-builder-error.zip](https://github.com/quarkusio/quarkus/files/9113696/graphql-schema-builder-error.zip)

### Steps to reproduce
- run the attached application
- Observe the error
- Comment out the query `failsToStartWithList` 
- Observe no error through GraphiQL(cf. screenshot below)
<img width="909" alt="Screen Shot 2022-07-14 at 12 44 27 PM" src="https://user-images.githubusercontent.com/29870258/179034847-3cb63de7-669b-4352-9e29-7a15a3d8f8d7.png">

### Output of `uname -a` or `ver`

Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000 arm64

### Output of `java -version`

openjdk version "17.0.1" 2021-10-19 LTS OpenJDK Runtime Environment Corretto-17.0.1.12.1 (build 17.0.1+12-LTS) OpenJDK 64-Bit Server VM Corretto-17.0.1.12.1 (build 17.0.1+12-LTS, mixed mode, sharing)

### Quarkus version or git rev

2.10.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

------------------------------------------------------------ Gradle 7.4.2 ------------------------------------------------------------  Build time:   2022-03-31 15:25:29 UTC Revision:     540473b8118064efcc264694cbcaa4b677f61041  Kotlin:       1.5.31 Groovy:       3.0.9 Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021 JVM:          17.0.1 (Amazon.com Inc. 17.0.1+12-LTS) OS:           Mac OS X 12.3.1 x86_64

### Additional information

Kotlin version: 1.6.21

**Comment by phillip-kruger** [comment]
Does this work when you use Java ?

**Comment by lcareto** [comment]
> Does this work when you use Java ?

@phillip-kruger , there's no issue when using java, even when mixing Kotlin data classes or enums with Java function, it works. 
The "failed to start" error seems to be specific to `Kotlin List`.

**Comment by phillip-kruger** [comment]
This will be fixed in the SmallRye 1.6.1 release.
